src/main/scala/mesosphere/marathon/core/task/tracker/impl/TaskOpProcessorImpl.scala

```diff
@@ -39,40 +40,55 @@ private[tracker] object TaskOpProcessorImpl {
       implicit ec: ExecutionContext): Future[Action] = {
       directTaskTracker.task(taskId).map {
         case Some(existingTask) =>
-          resolveForExistingTask(existingTask, status)
+          actionForTaskAndStatus(existingTask, status)
         case None =>
           Action.Fail(new IllegalStateException(s"$taskId of app [${taskId.appId}] does not exist"))
       }
     }
 
+    private[this] def actionForTaskAndStatus(task: Task, statusUpdate: TaskStatus): Action = {
+      //Step 1: if a resident task is terminated, it has to be handled different
+      resolveForResidentFailed(task, statusUpdate)
+        //Step 2: update existing running task
+        .orElse(resolveForLaunchedTask(task, statusUpdate))
+        //Step 3: nothing of the above: do nothing
+        .getOrElse {
+          log.warn("Got update for task which wasn't launched yet: {}", statusUpdate)
+          Action.Noop
+        }
+    }
+    private[this] def resolveForResidentFailed(task: Task, statusUpdate: TaskStatus): Option[Action] = {
+      for {
+        reservation <- task.reservationWithVolumes
+        launched <- task.launched
+        if Terminated.isTerminated(statusUpdate.getState)
+      } yield Unlaunch(task)
```

Comment:
Please update the scaladoc on line 37 – it should mention the possibility of this action as well.